### PR TITLE
Resolving issue  #1459

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR is to address issue  #1459.
I have also added a new function `create_list()` which will process a list of observations and return lists of counts, background and exposure maps.
I think this is useful when the user wants to analyse each run individually (or maybe do some grouping later...)

I guess that the names `run` and `create_list` may not be very nice.
Shall change them if required.

I still have a function`process_obs` to keep compatibility with the old `MapMaker`. Maybe this can be removed in the later versions?

@mackaiver : Do you like the present implementation?

There will be some merge conflicts with @leajouvin PR #1558 , so @cdeil and @registerrier , please merge either one so that the other can be edited...

### CODE REVIEW

**Comment by cdeil** [comment]
@AtreyeeS - Thanks!

There's a merge conflict, this has to be resolved or we can't put this in.

About the list: I'm not sure it's useful.
If you want per-run maps, you could just do this, no?
```
for obs in observations:
    maps_obs = MapMaker(...).run().results
```
(not literally, but roughly)

File: gammapy/cube/new.py [hunk 1]
```diff
@@ -287,13 +287,97 @@ def process_obs(self, obs):
...
+        maps: dict of stacked counts, background and exposure maps.
+        """
+
+        from astropy.utils.console import ProgressBar
+
+        for obs in ProgressBar(obs_list):
+            cs, count, exp, back = self.process_single_obs(obs)
+            self._add_cutouts(cs, count, exp, back)
+
+        maps = {
```
Comment (cdeil) [inline gammapy/cube/new.py]: Make this a property called e.g. "results" or "maps"?
Then callers could access it after processing some obs, or at the end.
You could still return it from `run` if you want.

**Comment by AtreyeeS** [comment]
I think it is useful to have a `create_list()`.
It can obviously be done by the user like you suggested, but `ProgressBar` needs a list, so the user has to `obs = [obs,]` and then take extract the list of counts and background maps... This is a few lines of code for each user then.

**Comment by AtreyeeS** [comment]
I am closing this and opening PR #1562

**Comment by cdeil** [comment]
> I think it is useful to have a create_list()

Besides stacking all runs, the next most common thing is to have per-run or per-group-of-runs maps, plus PSF and EDISP, and then to do a joint fit across those.

But for that your `create_lists` is not a great API, it probably requires introducing a `MapDataset` class that hast the maps and PSF / EDISP, and adapting both the MapMaker, MapEvaluator and MapFit.

@registerrier - I'll leave the decision whether adding `create_list` should go in for now or not.
I'm -1, but if it gets improved in the coming days I don't care much. I just don't think it's an API we should offer in a stable release (upcoming v0.8).

**Pull request closed without merging**